lib/ios-log.js

```diff
@@ -0,0 +1,235 @@
+
+import spawn from 'win-spawn';
+import through from 'through';
+import path from 'path';
+import fs from 'fs';
+import _ from 'lodash';
+import which from 'which';
+import glob from 'glob';
+import { getLogger } from 'appium-logger';
+import mkdirp from 'mkdirp';
+import touch from 'touch';
+import xcode from 'xcode'; 
+
+// Date-Utils: Polyfills for the Date object
+
+let START_TIMEOUT = 10000;
+let logger = getLogger('iOS-log');
+
+class IosLog { 
+  constructor(opts) {
+    this.udid = opts.udid;
+    this.simUdid = opts.simUdid;
+    this.showLogs = opts.showLogs;
+    this.proc = null;
+    this.onIosLogStart = null;
+    this.iosLogStarted = false;
+    this.iosLogStartTime = null;
+    this.calledBack = false;
+    this.loggingModeOn = true;
+    this.logs = [];
+    this.logRow = '';
+    this.logsSinceLastRequest = [];
+    this.startTimer = null;
+  }
+  async startCapture () {
+    if (this.udid) { 
+      this.loggingModeOn = false;
+      let spawnEnv = _.clone(process.env);
+      logger.debug('Attempting iOS device log capture via libimobiledevice idevicesyslog');
+      let err = await which('idevicesyslog');
+      if (!err) { 
+        this.proc = spawn('idevicesyslog', {env: spawnEnv});
+      } else {
+        logger.warn('Could not capture device log using libimobiledevice idevicesyslog.' +
```

Comment:
this might be a complicated change, but we probably want to use `SubProcess` instead of `spawn`, since it it's more 'async/await' friendly